services/provider/server/server.go

```diff
@@ -367,6 +370,37 @@ func (s *OCSProviderServer) getExternalResources(ctx context.Context, consumerRe
 		}),
 	})
 
+	if consumerResource.Spec.StorageQuotaInGiB > 0 {
+		hardLimit := resource.MustParse(fmt.Sprintf("%d%s", consumerResource.Spec.StorageQuotaInGiB, "Gi"))
+		clusterResourceQuota := &quotav1.ClusterResourceQuota{
+			ObjectMeta: metav1.ObjectMeta{Name: "ocs-storagecluster-quota"},
+			Spec: quotav1.ClusterResourceQuotaSpec{
+				Selector: quotav1.ClusterResourceQuotaSelector{
+					LabelSelector: &metav1.LabelSelector{
+						MatchExpressions: []metav1.LabelSelectorRequirement{
+							{
+								Key:      "dummy",
+								Values:   []string{"dummy"},
+								Operator: metav1.LabelSelectorOpNotIn,
+							},
+						},
+					},
+				},
+				Quota: corev1.ResourceQuotaSpec{
+					Hard: corev1.ResourceList{"requests.storage": hardLimit},
+				},
+			},
+		}
+
+		extR = append(extR, &pb.ExternalResource{
+			Name: "QuotaForConsumer",
+			Kind: "ClusterResourceQuota",
+			Data: mustMarshal(map[string]string{
+				"clusterResourceQuota": fmt.Sprintf("%+v\n", clusterResourceQuota),
+			}),
+		})
```

Comment:
Taking all of the comments above into account the entire thing should become much simpler:

```suggestion
		extR = append(extR, &pb.ExternalResource{
			Name: "QuotaForConsumer",
			Kind: "ClusterResourceQuota",
			Data: mustMarshal(quotav1.ClusterResourceQuotaSpec{
				Quota: corev1.ResourceQuotaSpec{
					Hard: corev1.ResourceList{
						"requests.storage": resource.NewScaledQuantity(
							consumerResource.Spec.StorageQuotaInGiB,
							resource.Giga,
						),
					},
				},
			},),
		})
```